### ISSUE TITLE
Terraform remote workspace version preventing local execution

### ISSUE THREAD

**Issue by taiidani** [hashicorp/terraform#27541]
### Terraform Version
```
Terraform v0.14.4
+ provider registry.terraform.io/digitalocean/digitalocean v1.22.2
+ provider registry.terraform.io/hashicorp/archive v1.3.0
+ provider registry.terraform.io/hashicorp/aws v3.5.0
+ provider registry.terraform.io/hashicorp/http v1.2.0
+ provider registry.terraform.io/hashicorp/null v2.1.2
```

### Terraform Configuration Files
```terraform
terraform {
  # ...

  backend "remote" {
    hostname     = "app.terraform.io"
    organization = "rnd"

    workspaces {
      name = "infra"
    }
  }
}
```

### Debug Output

https://gist.github.com/taiidani/cd3a4bf53cabce1ad02e426faf977cfa

### Expected Behavior

Able to import new items into my state file.

### Actual Behavior

During my first `apply` on a machine, adding a single `aws_secrets_manager` secret. It failed with:

```
Error: Provider produced inconsistent result after apply

When applying changes to
module.aws.aws_secretsmanager_secret.lil_dumpster_discord_token, provider
"registry.terraform.io/hashicorp/aws" produced an unexpected new value: Root
resource was present, but now absent.

This is a bug in the provider, which should be reported in the provider's own
issue tracker.
```

I then tried to import it and received the following error:
```
Error: Error loading state: Remote workspace Terraform version "0.12.9" does not match local Terraform version "0.14.4"
```
I have been unable to get Terraform to work around this.

### Additional Context

![image](https://user-images.githubusercontent.com/902500/104997170-fbad1180-59dd-11eb-92f6-2afb9ed2a7e1.png)

I believe this is related to my Terraform Cloud configuration. Namely that I have it set to "local" execution _but_ I used to have it set to remote execution a year ago. The output of my last run from a year ago:

```
Terraform v0.12.9
Configuring remote state backend...
Initializing Terraform configuration...
2019/10/12 16:07:26 [DEBUG] Using modified User-Agent: Terraform/0.12.9 TFC/8f02cbea59
```

The current state file is definitely configured against 0.14.4:
```
✗ terraform state pull | head
{
  "version": 4,
  "terraform_version": "0.14.4",
  "serial": 158,
  "lineage": "78a7c3d7-05b1-c941-8d5b-d665e479cb6b",
  "outputs": {},
  "resources": [
    {
      "module": "module.aws",
      "mode": "managed",
```

**Comment by billinghamj** [comment]
Similar bug here:

```
Error: Error loading state: Remote workspace Terraform version "0.12.20" does not match local Terraform version "0.14.4"
```

Maybe something's been broken on Terraform Cloud's backend?

**Comment by qwerty1979bg** [comment]
This was supposed to be fixed with https://github.com/hashicorp/terraform/pull/27407 🧐

@billinghamj billinghamj Does the workaround proposed in https://github.com/hashicorp/terraform/issues/27353 works for you ?

> ... using the -ignore-remote-version flag with terraform import  ...

**Comment by alisdair** [comment]
Thanks for reporting this. This is a duplicate of #27480, fixed in #27498, and the fix will be in the upcoming 0.14.5 release of Terraform.

Using the `-ignore-remote-version` flag on these commands should work until then. There's also a workaround which allows you to skip the flag:

- Visit the workspace settings page on Terraform Cloud
- Change the execution mode to remote, and save settings
- Change the Terraform version to match your local Terraform version, and save settings
- Change the execution mode back to local, and save settings

Sorry for the inconvenience!

**Comment by billinghamj** [comment]
Is there an ETA for 0.14.5?

**Comment by alisdair** [comment]
We currently expect to release 0.14.5 on Wednesday 20th January (tomorrow), unless something unforeseen causes a delay.

**Comment by andre-lx** [comment]
Hi. 

In our case, we need to change the `Execution mode` from `local` to `remote` 
Update the terraform version
Save the settings 

And then the remote workspace terraform version match the local terraform version.

**Comment by taiidani** [comment]
@alisdair Interesting. So I've upgraded to 0.14.5 and tried toggling "local" to "remote" and back so I could update the version. I got through the version discrepancy issue, but it still looks like it's comparing against the (disabled) Remote configuration:

```
terraform import module.aws.aws_secretsmanager_secret.lil_dumpster_discord_token arn:aws:secretsmanager:us-west-2:597609949659:secret:lil-dumpster/discord-token-SyzUB5
Releasing state lock. This may take a few moments...

Warning: Value for var.consul_encryption_key unavailable

The value of variable "consul_encryption_key" is marked as sensitive in the
remote workspace. This operation always runs locally, so the value for that
variable is not available.


Warning: Value for var.nomad_encryption_key unavailable

The value of variable "nomad_encryption_key" is marked as sensitive in the
remote workspace. This operation always runs locally, so the value for that
variable is not available.


Error: Value for undeclared variable

A variable named "consul_encryption_key" was assigned a value, but the root
module does not declare a variable of that name. To use this value, add a
"variable" block to the configuration.


Error: Value for undeclared variable

A variable named "nomad_encryption_key" was assigned a value, but the root
module does not declare a variable of that name. To use this value, add a
"variable" block to the configuration.


Error: Value for undeclared variable

A variable named "ssh_public_key" was assigned a value, but the root module
does not declare a variable of that name. To use this value, add a "variable"
block to the configuration.
```

I used to use variables in my Terraform Cloud configuration, but no longer use them. Both the latest state in Terraform Cloud and on my local have no mention of these variables (and, in fact, no consul or nomad resources at all anymore), but yet it's displaying these warnings and errors as if it were running with Remote state turned on. This _only_ appears when I run an `import`. Both `plan` and `apply` are fully succeeding with clean output.

To get around this I had to again go into Terraform Cloud, switch the settings to Remote state, delete the variables from the UI, then switch back to Local execution. So I managed to get through the block! I'm just very confused why I had to to this when it was specifically set to run Local?

**Comment by andre-lx** [comment]
> I'm just very confused why I had to to this when it was specifically set to run Local?

This is the important part actually. Since the execution mode is local, why terraform is comparing the local files against the remote execution mode variables and version.

**Comment by alisdair** [comment]
> This is the important part actually. Since the execution mode is local, why terraform is comparing the local files against the remote execution mode variables and version.

@andre-lx This is not the intended behaviour. It was a bug, which should now be fixed.

**Comment by alisdair** [comment]
@taiidani I'm sorry to hear about that problem, but I'm glad you were able to unblock yourself! That sounds like a separate issue from the workspace version check bug, and a more general case of the local state migration struggling with Terraform Cloud remote vs local operations mode.

In future, if something like this happens, a common "fix" is to clear your local `.terraform` directory inside the configuration working dir. This removes the local cached backend setup, and doing so avoids the state migration code, which has a few obscure bugs.

There is a related proposal not to run state migration in #18011.
